### ISSUE TITLE
Dependabot does not completely ignore Node.js dependency and fails when the dependency is private and on a registry that requires VPN access

### ISSUE THREAD

**Issue by msschambach** [dependabot/dependabot-core#10663]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package ecosystem

npm

### Package manager version

pnpm >=9.0.0

### Language version

Node.js >=20.9.0

### dependabot.yml content

version: 2
updates:
  - package-ecosystem: "npm" # See documentation for possible values
    directory: "/" # Location of package manifests
    schedule:
      interval: "daily"
      time: "10:00"
      timezone: "Etc/UTC"
    ignore:
      - dependency-name: "<REDACTED>-api-client"

### What you expected to see, versus what you actually saw

I expect for dependabot to completely ignore the ignored dependency and for all dependabot actions to pass. What happens however is that dependabot somehow is still trying to fetch the ignored dependency. See screenshots below:
 
 
![Screenshot 2024-09-24 at 10 51 15](https://github.com/user-attachments/assets/5da4cd15-eeab-4e7d-a4f2-a270839c2c3c)
![Screenshot 2024-09-24 at 10 52 53](https://github.com/user-attachments/assets/ac86561e-e306-44c6-a6cb-202edbd25699)

The dependency in question is a private package in a private nexus registry in my organisation. This registry however is only accessible via a VPN tunnel which is perfectly fine in our custom github action servers that have all the necessary config to access the registry, but dependabot doesn't have that access. 

I know that dependabot [can now be managed on self-hosted runners](https://docs.github.com/en/code-security/dependabot/working-with-dependabot/managing-dependabot-on-self-hosted-runners) which should solve this issue for our organisation but we are not ready to do that yet due to the effort involved.

What I expect is that ignored dependencies should be completely ignored but that is not the case. 

I've also seen a similar issue to this here: https://github.com/dependabot/dependabot-core/issues/10620

I've added an example manifest for reproducing the issue. You'll need to replace `@myorg/example-private-vpn-access-npm-repo` with a private package in a registry that requires VPN access. I've already tried reproducing this issue with a private package on Githubs' NPM registry but wasn't able to here: https://github.com/msschambach/dependabot-ignore-private-repo-pnpm

**TLDR:** It's only an issue if the private repository requires **_VPN access_**. If it only requires auth authentication then the dependabot builds work even if the auth token isn't provided.

### Smallest manifest that reproduces the issue

{
  "name": "dependabot-ignore-private-repo-pnpm",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "engineStrict": true,
  "engines": {
    "node": ">=20.9.0",
    "pnpm": ">=9.0.0"
  },
  "scripts": {
    "preinstall": "npx only-allow pnpm",
    "dev": "vite",
    "build": "vite build",
    "lint": "eslint .",
    "preview": "vite preview",
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "dependencies": {
    "@myorg/example-private-vpn-access-npm-repo": "1.0.0",
    "@vitejs/plugin-react": "^4.3.1",
    "eslint": "^9.11.1",
    "eslint-plugin-react": "^7.36.1",
    "eslint-plugin-react-refresh": "^0.4.12",
    "react": "^18.3.1",
    "react-dom": "^18.3.1",
    "vite": "^5.4.7"
  }
}

**Comment by htndev** [comment]
Same here. Using wildcard for certain packages in `ignore` section doesn't help. In logs, I see the same message `"ignored, no update allowed"`, but down below it keeps attempting to install dependencies, which fail the entire update process. Looking forward to seeing a fix for it

**Comment by v-thavaahariharangit** [comment]
With ignore condition: https://github.com/dsp-testing/pnpm-private-registry-issue-10663/actions/runs/16938707657/job/48001675786

<img width="1074" height="316" alt="Image" src="https://github.com/user-attachments/assets/2212608c-483b-44ad-a132-64d974112e0b" />

With out ignore condition: https://github.com/dsp-testing/pnpm-private-registry-success-with-lock

Dependabot.yml used above
```
version: 2
updates:
  - package-ecosystem: "npm"
    directory: "/"
    versioning-strategy: "increase"
    schedule:
      interval: "daily"
    registries:
      - jfrog-artifactroy
    ignore:
      - dependency-name: "lod*"
```
Ref: https://github.com/dsp-testing/pnpm-private-registry-issue-10663/blob/main/.github/dependabot.yml


Please consider using `*api-client` if there is pre text.
